src/TextDrawingComponent.cpp

```diff
@@ -0,0 +1,316 @@
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::PositionMiddle:
+				{
+					retVal = Justification::centredLeft;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::PositionBottom:
+				{
+					retVal = Justification::bottomLeft;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::Reserved:
+				default:
+					break;
+			}
+		}
+		break;
+
+		case isobus::TextualVTObject::HorizontalJustification::PositionMiddle:
+		{
+			switch (verticalJustification)
+			{
+				case isobus::TextualVTObject::VerticalJustification::PositionTop:
+				{
+					retVal = Justification::centredTop;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::PositionMiddle:
+				{
+					retVal = Justification::centred;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::PositionBottom:
+				{
+					retVal = Justification::centredBottom;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::Reserved:
+				default:
+					break;
+			}
+		}
+		break;
+
+		case isobus::TextualVTObject::HorizontalJustification::PositionRight:
+		{
+			switch (verticalJustification)
+			{
+				case isobus::TextualVTObject::VerticalJustification::PositionTop:
+				{
+					retVal = Justification::topRight;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::PositionMiddle:
+				{
+					retVal = Justification::centredRight;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::PositionBottom:
+				{
+					retVal = Justification::bottomRight;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::Reserved:
+				default:
+					break;
+			}
+		}
+		break;
+
+		default:
+		{
+		}
+		break;
+	}
+	return retVal;
+}
+
+std::uint8_t TextDrawingComponent::prepare_text_painting(Graphics &g,
+                                                         std::shared_ptr<isobus::FontAttributes> font,
+                                                         char referenceCharForWidthCalc)
+{
+	Font juceFont = Font(Font::getDefaultMonospacedFontName(), 0, 0);
+	std::uint8_t fontHeight;
+	auto fontType = font->get_type();
+	auto colour = parentWorkingSet->get_colour(font->get_colour());
+	int fontStyleFlags = Font::FontStyleFlags::plain;
+
+	if (font->get_style(isobus::FontAttributes::FontStyleBits::Bold))
+	{
+		fontStyleFlags |= Font::FontStyleFlags::bold;
+	}
+
+	if (font->get_style(isobus::FontAttributes::FontStyleBits::Italic))
+	{
+		fontStyleFlags |= Font::FontStyleFlags::italic;
+	}
+
+	if (font->get_style(isobus::FontAttributes::FontStyleBits::Underlined))
+	{
+		fontStyleFlags |= Font::FontStyleFlags::underlined;
+	}
+
+	juceFont.setStyleFlags(fontStyleFlags);
+	juceFont.setHeight(font->get_font_height_pixels());
+
+	auto fontWidth = juceFont.getStringWidthFloat(juce::String::fromUTF8(&referenceCharForWidthCalc, 1));
+	fontHeight = font->get_font_width_pixels();
```

Comment:
@ad3154  I also spotted a weird thing here: I assume here the get_font_height_pixels() should have been called as it will be used for line count calculation later on. 

It was the same in the InputString:
https://github.com/Open-Agriculture/AgIsoVirtualTerminal/blob/main/src/InputStringComponent.cpp#L66
and the OutputString classes:
https://github.com/Open-Agriculture/AgIsoVirtualTerminal/blob/main/src/OutputStringComponent.cpp#L73

What do you think was it intentional or just a typo?